episode4/README.md

```diff
@@ -1,64 +1,400 @@
+the wrath of Chtulu.
+
+So, let's build a (temporary) test to demonstrate the differences. Tests are not only great ways to
+verify your code, but also to experiment. Just make sure start without any other changes in your git
+repository. This will make it very easy to go back.
+
+In `test/calculator.spec.js` add the following code:
+
+```js
+it.only("can access the result", () => {
+  calculator.add(2)
+
+  expect(calculator._result).to.eql(2)
+  expect(calculator.result()).to.eql(2)
+  // expect(calculator.result).to.eql(2)
+});
+```
+
+There are three things to note.
+
+The first one is that the test starts with `it.only`. This means that
+it will be the only test that runs. The second one is that last assertion is commented out. Try and
+see what happens when you remove the comment.
+
+Lastly, the first assertion shows that the `_result` attribute in the calculator object is exposed
+and theoretically can be used outside the calculator object. But just because you can doesn't mean
+that you should.
+
+So what happens when we change the implementation of the result function to this:
+
+```js
+get result() {
+  return this._result;
+}
+```
+
+Which assertions do you have to comment / uncomment to still keep the test passing?
+
+Since ES5 there is a way to define getters and setters. This will make the name of the method available
+to be accessed as an attribute. So that the above example of the `result` getter can be accessed as:
+
+```js
+calculator.result
+```
+
+The use of getters and setters on *properties* of a class is common.
+
+The lesson here is that there are multiple ways and each has it's advantages and drawbacks.
+Experiment with your code. Even if you have decided to go one way initially, it doesn't mean that
+you cannot change your mind later!
+
+With tests and making tiny changes it should be relatively easy to go in a different direction.
+
+## Exercise: Refactor to getter
+Now refactor our code and test to use the getter. We would take tiny steps, changing it one test at
+the time and see the first one pass. Then move on to the next. Try to avoid making a big change.
+
+Hint: Revert your code to the point before the experimentation and make sure all tests are running.
+
+## Exercise: Implement other operations
+Add the other operations to the calculator class using TDD. Make sure you can chain the methods.
+
+- Subtraction (minus)
+- Multiplication (times)
+- Square (square)
+
+What if you implement one operation using a getter from the beginning? And what if you don't? Does
+it have a large impact? Experiment with that.
+
+When you are finished, you can add the following regression test. Is this passing? If not, find out
+why and what test you have missed in your implementation of the operations.
+
+```js
+describe("Regression test", () => {
+	it("can combine multiple chained commands", () => {
+		calculator
+			.add(7)
+			.subtract(2)
+			.square()
+			.multiply(4)
+
+		expect(calculator.result).to.eql(100)
```

Comment:
`7 - 2 ^ 2 * 4 = -9`

Pure technically speaking. But we haven't covered the lazy part of coming to the result yet. I'm gonna write a note, cus it bothers me. :P